admin/pre-reg/views.py

```diff
@@ -0,0 +1,171 @@
+        'admin': 'admin_placeholder'
+    }
+    #reviewers = get_prereg_users()
+    reviewers = ['admin_placeholder']
+
+    #context = {'user_info': user, 'reviewers': reviewers, 'user': request.user}
+    context = {
+        'user_info': user,
+        'reviewers': reviewers
+    }
+    return render(request, 'pre-reg/prereg.html', context)
+
+@login_required
+# @user_passes_test(is_in_prereg_group)
+def prereg_form(request, draft_pk):
+    """Redirects to prereg form review page if user has prereg access
+    :param draft_pk: Unique id for selected draft
+    :return: Redirect to prereg form review page with draft obj
+    """
+    draft = get_draft_or_error(draft_pk)
+    context = {
+        'draft': utils.serialize_draft_registration(draft)
+    }
+    return render(request, 'pre-reg/edit_draft_registration.html', context)
+
+@login_required
+# @user_passes_test(is_in_prereg_group)
+def get_drafts(request):
+    """Determines whether a user is in the general_administrator_group
+    :param user: User wanting access to administrator material
+    :return: True if general administrator False if not
+    """
+    prereg_schema = MetaSchema.find_one(
+        Q('name', 'eq', 'Prereg Challenge') &
+        Q('schema_version', 'eq', 2)
+    )
+    all_drafts = DraftRegistration.find(
+        Q('registration_schema', 'eq', prereg_schema) &
+        Q('approval', 'ne', None)
+    )
+    serialized_drafts = {
+        'drafts': [utils.serialize_draft_registration(d) for d in all_drafts]
+    }
+    return JsonResponse(
+        serialized_drafts
+    )
+
+@login_required
+# @user_passes_test(is_in_prereg_group)
+@csrf_exempt
+def approve_draft(request, draft_pk):
+    """Approves current draft
+    :param user: Current logged in user
+    :param draft_pk: Unique id for current draft
+    :return: DraftRegistrationApproval obj
+    """
+    draft = get_draft_or_error(draft_pk)
+
+    user = load_osf_user(request.user)
+    try:
+        draft.approve(user)
+    except PermissionsError:
+        draft.approval.add_authorizer(user, save=True)
+        draft.approve(user)
+    return JsonResponse({})
+
+@login_required
+# @user_passes_test(is_in_prereg_group)
+@csrf_exempt
+def reject_draft(request, draft_pk):
+    """Rejects current draft
+    :param user: Current logged in user
+    :param draft_pk: Unique id for current draft
+    :return: DraftRegistrationApproval obj
+    """
+    draft = get_draft_or_error(draft_pk)
+
+    user = load_osf_user(request.user)
+    try:
+        draft.reject(user)
+    except PermissionsError:
+        draft.approval.add_authorizer(user, save=True)
+        draft.reject(user)
+    return JsonResponse({})
+
+
+@login_required
+# @user_passes_test(is_in_prereg_group)
+@csrf_exempt
+def update_draft(request, draft_pk):
+    """Updates current draft to save admin comments
+
+    :param draft_pk: Unique id for current draft
+    :return: DraftRegistration obj
+    """
+    data = json.load(request)
```

Comment:
It's working but this should probably be `json.loads(request.body)`. I believe it's working on an edge case around the request.\__repr__ method returning the request.body.